Recipes/AsyncContext.md

```diff
@@ -37,54 +37,61 @@ Task<Void, Never> {
 You need to kick off some work from the main thread, complete it in the background, and then update some state back on the main thread.
 
 ```swift
-beforeWorkBegins()
-
-DispatchQueue.global.async {
-    let possibleResult = expensiveWork(arguments)
-    
-    DispatchQueue.main.async {
-        afterWorkIsDone(possibleResult)
+func doWork() {}
+    // We assume we're on the main thread here
+    beforeWorkBegins()
+
+    DispatchQueue.global.async {
+        let possibleResult = expensiveWork(arguments)
+        
+        DispatchQueue.main.async {
+            afterWorkIsDone(possibleResult)
+        }
     }
 }
 ```
 
-### Solution #1: the work can be made async
+### Solution #1: Someone wrote an `async` wrapper for `expensiveWork()`
 
-Assumptions: both `beforeWorkBegins` and `afterWorkIsDone` are `MainActor`-isolated, and we are starting on the main thread
+If there is an `async` wrapper for `expensiveWork()`, the "background work" recipe is simpler in the world of structured concurrency. This recipe assumes that both `arguments` and `result` are `Sendable` (safe to cross isolation domains). Note this was true in the dispatch-queue based world, too!
 
 ```swift
-// hazard 1: timing. Does this call happen here, or within the Task?
-beforeWorkBegins()
+@MainActor
+func doWork() async {
+    beforeWorkBegins()
```

Comment:
You have moved `beforeWorkBegins` out of the synchronous calling context into the async one. This could be totally fine! And it also could not, as it is now affected by Task ordering.